iocore/net/quic/QUICApplication.cc

```diff
@@ -138,38 +138,31 @@ QUICStreamIO::write(IOBufferReader *r, int64_t len)
 {
   SCOPED_MUTEX_LOCK(lock, this->_write_vio->mutex, this_ethread());
 
-  int64_t bytes_avail = this->_write_buffer->write_avail();
-
-  if (bytes_avail > 0) {
-    if (is_debug_tag_set(tag_stream_io)) {
-      if (this->_write_vio->nbytes == INT64_MAX) {
-        QUICStreamIODebug("nbytes=- ndone=%" PRId64 " write_avail=%" PRId64 " write_len=%" PRId64, this->_write_vio->ndone,
-                          bytes_avail, len);
-      } else {
-        QUICStreamIODebug("nbytes=%" PRId64 " ndone=%" PRId64 " write_avail=%" PRId64 " write_len=%" PRId64,
-                          this->_write_vio->nbytes, this->_write_vio->ndone, bytes_avail, len);
-      }
-    }
-
-    int64_t bytes_len = std::min(bytes_avail, len);
-    int64_t nwritten  = this->_write_buffer->write(r, bytes_len);
-
-    if (nwritten > 0) {
-      this->_nwritten += nwritten;
-    }
-
-    return nwritten;
-  } else {
-    return 0;
+  int64_t written = this->_write_buffer->write(r, len);
+  if (written > 0) {
+    this->_nwritten += written;
   }
+
+  return written;
 }
 
-// TODO: Similar to other "write" apis, but do not copy.
 int64_t
 QUICStreamIO::write(IOBufferBlock *b)
 {
-  ink_assert(!"not implemented yet");
-  return 0;
+  MIOBuffer buffer;
+  IOBufferReader *reader = buffer.alloc_reader();
+
+  buffer.append_block2(b);
+
+  // TODO: Remove this read_avail
+  int64_t written = reader->read_avail();
+  if (written == 0) {
```

Comment:
No idea how to remove this read_avail

Reply:
If that's the condition, you can change it to `is_read_avail_more_than(0)`.